localstack/aws/handlers/cors.py

```diff
@@ -229,22 +230,29 @@ def __call__(self, chain: HandlerChain, context: RequestContext, response: Respo
             if headers.get(header) == "":
                 del headers[header]
 
+        request_headers = context.request.headers
+        # CORS headers should only be returned when an Origin header is set.
         # use DISABLE_CORS_HEADERS to disable returning CORS headers entirely (more restrictive security setting)
         # also don't add CORS response headers if the service manages the CORS handling
-        if config.DISABLE_CORS_HEADERS or not should_enforce_self_managed_service(context):
+        if (
+            "Origin" not in request_headers
+            or config.DISABLE_CORS_HEADERS
+            or not should_enforce_self_managed_service(context)
+        ):
             return
 
-        request_headers = context.request.headers
         self.add_cors_headers(request_headers, response_headers=headers)
 
     @staticmethod
     def add_cors_headers(request_headers: Headers, response_headers: Headers):
         if ACL_ORIGIN not in response_headers:
             response_headers[ACL_ORIGIN] = (
-                request_headers["origin"]
-                if request_headers.get("origin") and not config.DISABLE_CORS_CHECKS
+                request_headers["Origin"]
+                if request_headers.get("Origin") and not config.DISABLE_CORS_CHECKS
                 else "*"
             )
+        if "*" not in response_headers.get(ACL_ORIGIN, ""):
+            response_headers[ACL_CREDENTIALS] = "true"
```

Comment:
Did we have any problems with that or is this just for completeness?

Reply:
I think I've seen the issue pops sometimes in Chrome or Safari where it would say that the request was not allowed because of this header missing. I believe it can be good to add it, as the browser could block some requests in the frontend if we don't include it.

> When a request's credentials mode ([Request.credentials](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials)) is include, browsers will only expose the response to the frontend JavaScript code if the Access-Control-Allow-Credentials value is true.

> If the CORS request includes credentials, the response must include the Access-Control-Allow-Credentials: true header, and the value of Access-Control-Allow-Origin must reflect the request's Origin header (* isn't an acceptable value if the request has credentials).

As we control the list of allowed domains, I believe this can be useful, maybe for cognito login page for example? 
We also could add a `CORS_ALLOW_CREDENTIALS` feature flag in the same way as `flask_cors`, to add this header only if it's set?

Reply:
@dfangl do you think we should do something like above, or just merge this PR?